eclair-core/src/main/scala/fr/acinq/eclair/channel/Channel.scala

```diff
@@ -2448,6 +2451,9 @@ class Channel(val nodeParams: NodeParams, val wallet: EclairWallet, remoteNodeId
       try {
         s(event)
       } catch {
+        case t: SQLException =>
+          log.error(t, "fatal database error\n")
+          sys.exit(-2)
```

Comment:
Can we test that (probably not easy because of `sys.exit`, unless we somehow mock it)?
Are we sure that all DB errors will be wrapped in `SQLException`s?

Reply:
I did test it manually by making the db call fail after 3 calls, and running one of the `NormalStateSpec`

Reply:
Another way would be to use the same mechanism as for postgres lock:
https://github.com/ACINQ/eclair/blob/master/eclair-core/src/main/scala/fr/acinq/eclair/db/pg/PgUtils.scala#L53-L84

But that would mean add that handler to the `ChannelFactory`, maybe a bit overkill.